### DISCUSSION TITLE
Private modules

### DISCUSSION THREAD

**Discussion by shykes** [dagger/dagger#12284] [Ideas]
## Problem

Sometimes your Dagger module is publicly reachable, but not intended to be publicly used. There is no way to communicate this, so private modules pollute daggerverse search results, and users get no warning when installing them.

## Solution

- Add an optional `private` boolean flag to `dagger.json`
- `dagger init --private` sets the flag
- Daggerverse honors the flag by hiding (partially or totally) private modules in search results 
- `dagger install` returns an error on private modules, overrideable with `-f`

**Comment by shykes** [comment]
cc @sagikazarmark 😁

**Comment by sagikazarmark** [comment]
Yes, please!

Daggerverse should hide already visible modules.

I wonder if it would make sense to create a daggerverse specific section in dagger.json for future settings.

Private may also be a misleading name: my Open Source project CI isn't private, it's just not reusble. Same for my test modules.

So I'm thinking daggerverse.publish = false.

**Comment by helderco** [comment]
`package.json` (node) uses `private` too: https://docs.npmjs.com/cli/v10/configuring-npm/package-json#private.

I think it's reasonable to consider `private` to mean it's not meant to be shared or reused, even if the source code is open.

If the code is in a private repo then the `private` config here isn't even necessary.
